src/third_party/progress/progress.hpp

```diff
@@ -152,45 +164,46 @@ class C_Progress
 class C_Progress_display : public C_Progress
 {
   public:
+      C_Progress_display()
+          : m_os(std::cout), m_msg("\n") 
+      {
+      }
     /** @brief Standard constructor
       * @param expected_count The number of step of the process
       * @param os the stream that will be used for display
-      * @param s1 the starting string
-      * @param s2 String used before the |--| sequence
-      * @param s3 String used after the  |--| sequence
+      * @param msg the status string
       **/
     explicit C_Progress_display ( unsigned long ulExpected_count,
                                      std::ostream & os = std::cout,
-                                     const std::string & s1 = "\n", //leading strings
-                                     const std::string & s2 = "",
-                                     const std::string & s3 = "" )
+                                     const std::string & msg = "\n")
     // os is hint; implementation may ignore, particularly in embedded systems
-    : m_os ( os ), m_s1 ( s1 ), m_s2 ( s2 ), m_s3 ( s3 )  { restart ( ulExpected_count ); }
+    : m_os ( os ), m_msg (msg){ restart ( ulExpected_count, msg); }
 
     /** @brief Initializer of the C_Progress_display class
      * @param expected_count The number of step of the process
+     * @param msg the status string
      **/
-    void restart ( unsigned long ulExpected_count )
+    void restart ( unsigned long ulExpected_count, const std::string& msg=std::string()) override
     //  Effects: display appropriate scale
     //  Postconditions: count()==0, expected_count()==expected_count
     {
-      C_Progress::restart ( ulExpected_count ); //-- Initialize the base class
-
-      m_os << m_s1 << "0%   10   20   30   40   50   60   70   80   90   100%\n"
-      << m_s2 << "|----|----|----|----|----|----|----|----|----|----|"
-      << std::endl  // endl implies flush, which ensures display
-      << m_s3;
+      C_Progress::restart ( ulExpected_count,  msg); //-- Initialize the base class
+      if (!msg.empty())
+          m_msg = msg;
```

Comment:
I've added a comment that an empty string leaves the last status untouch. I don't think that an empty status makes sense semantically.